mod/enhanced_user_search/lib/sql.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+namespace NRC\EUS\SQL;
+
+class Constants {
+  public function get() {
+    $dbprefix = elgg_get_config("dbprefix");
+    $dbname = elgg_get_config("dbname");
+
+    $tableName = "NRC_EUS_mvMemberSearch";
+    $procName = "NRC_EUS_Refresh_MemberSearch";
+
+    $BUILD_SQL = "
+    CREATE TABLE IF NOT EXISTS `{$tableName}` (
+      user_guid bigint(20) unsigned not null,
+      name text not null,
+      contactemail text,
+      education text,
+      gc_skills text,
+      portfolio text,
+      work text
+    );
+    CREATE TABLE IF NOT EXISTS `{$tableName}_tmp` (
+      user_guid bigint(20) unsigned not null,
+      name text not null,
+      contactemail text,
+      education text,
+      gc_skills text,
+      portfolio text,
+      work text
+    );";
+
+    $INDEX_SQL = "
+    CREATE FULLTEXT INDEX NRC_EUS_MemberSearch ON
+      `{$tableName}`(
+        name,
+        contactemail,
+        education,
+        gc_skills,
+        portfolio,
+        work
+      );
+    CREATE FULLTEXT INDEX NRC_EUS_MemberSearchTmp ON
+      `{$tableName}_tmp`(
+        name,
+        contactemail,
+        education,
+        gc_skills,
+        portfolio,
+        work
+      );";
+
+    $REFRESH_PROC = "
+    CREATE PROCEDURE `{$procName}` ()
+    BEGIN
+      TRUNCATE TABLE `{$tableName}_tmp`;
+
+      INSERT INTO `{$tableName}_tmp`
+      SELECT
+        a.guid as user_guid,
+        a.name as name,
+        c.string as contactemail,
+        GROUP_CONCAT(DISTINCT f1.title SEPARATOR ', ') as education,
+        GROUP_CONCAT(DISTINCT f2.title SEPARATOR ', ') as gc_skills,
+        GROUP_CONCAT(DISTINCT f3.title SEPARATOR ', ') as portfolio,
+        GROUP_CONCAT(DISTINCT f4.title SEPARATOR ', ') as work
```

Comment:
@lucbelliveau  One of these group_concat seems to be erroring out when I try to run the update on preprod with "Row 1159 was cut by GROUP_CONCAT()".